docs/usecases/streaming.md

```diff
@@ -3,4 +3,105 @@ id: usecases_streaming
 title:  "Streaming"
 ---
 
-Coming soon...
+All material on this pages assumes knowledge about the `ZIO[R, E, A]` type.
+
+## Why we need streams?
+TBD
+
+## How `zio-streams` are build
+
+The backbone of `zio-streams` are 3 types: `ZStream`, `Pull` and `ZSink`. 
+`ZStream[R, E, A]` represents the stream itself, which produces effectful values of type `A` and might end up with 
```

Comment:
What is "effectful value" ?

Reply:
No problem, questions like this help me to write it in a more straight forward way.

I think the use of word 'effect' is unfortunate because of  confusion with side effects and those are different things, but well that's the terminology that people use so I'm sticking with it. 

Example:
We say that one can model absence of value when evaluating a function with 'Option'. And the fact that your function returns not plain A but Option[A] means your using effects to model computation. Another example would be using Either to model failure. So when you're returning Either[Error, A] from function you're modeling reporting errors with effect.

Back to your original question, our A is being created by modeling the computation with ZIO (allowing us to do whatever we want) so again we have an effect involved. Thus it is called effectful value as to get it we need to evaluate that effect.

Reply:
Thanks. Still, I would expect the same terminology in both docs (see: https://zio.dev/docs/datatypes/datatypes_stream).

Reply:
You mean this sentence?

"A Stream[E, A] represents an effectful stream that can produce values of type A, or potentially fail with a value of type E."?

That's a good point the docs should be consistent.
The word 'effect' is not a very well defined one I guess. When I was writing this I focused more on the internals and that each A comes from a separate evaluation. Will rephrase this.

Reply:
Could be phrased, `effectfully produced value`, i.e. effects are required for the production of the value.

Roughly speaking, "functional effect" is the functional programming version of "side-effect", but in FP libraries, "functional effect" tends to be truncated to "effect", to avoid tedious repetition.